### PR TITLE
[BREAKING] refactor!: remove theft reporting

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

Inline with the next steps in [this post-mortem](https://medium.com/interlay/kintsugi-released-urgent-security-patches-aebf969ee087) we will be removing theft reporting to significantly reduce the liquidation risk to Vaults in the system.

## Breaking Changes
- Removed `Relay` pallet and RPC
- Block submission extrinsics moved to `BTCRelay`

### CODE REVIEW

**Comment by gregdhill** [comment]
@nud3l @sander2 should we also remove `VaultStaus::CommittedTheft`? That will require a migration.

**Comment by sander2** [comment]
> @nud3l @sander2 should we also remove `VaultStaus::CommittedTheft`? That will require a migration.

I would like to remove it to reduce the number of edge cases that have to be dealt with. However, it _would_ mean that the vaults in the system that were liquidated due to theft will now show as `liquidated` in the UI, which may raise some questions

File: crates/btc-relay/src/lib.rs [hunk 1]
```diff
@@ -244,8 +328,6 @@ pub mod pallet {
         InvalidHeaderSize,
         /// Block already stored
         DuplicateBlock,
-        /// Block already stored and is not head
-        OutdatedBlock,
```
[the remark this answers is not in the document]
Comment (gregdhill) [reply crates/btc-relay/src/lib.rs:248]: That was when we had SLA scoring for relaying and still wanted to reward those who submit (late).

**Review by sander2 — DISMISSED** [review]
LGTM but maybe @nud3l (or someone else) should sign off on removing the CommittedTheft status since it'll have an influence on perception in the UI?

**Comment by gregdhill** [comment]
@sander2 I'll revert the migration for now since @nud3l is offline today and this is blocking my work on #605.

**Pull request merged**